components/Panel/ClaimPanel.tsx

```diff
@@ -18,14 +19,43 @@ export function ClaimPanel() {
   const { withdrawRewardsMutation, isWithdrawingRewards } = useWithdrawRewards("claim");
   const { withdrawAndRestakeMutation, isWithdrawingAndRestaking } = useWithdrawAndRestake("claim");
   const { outstandingRewards, getStakingDataFetching } = useStakingContext();
+  const notificationHandler = useContractInteractionNotifications();
   const isDelegate = getDelegationStatus() === "delegate";
 
   function withdrawRewards() {
-    withdrawRewardsMutation({ voting });
+    if (!outstandingRewards) return;
+
+    withdrawRewardsMutation(
```

Comment:
this is the pattern used for most of the mutations. you'll notice that he use `notificationHandler` in two different situations: for showing pending transactions (uses a `ContractTransaction`) and for showing settled (succeeded or failed) which uses a `ContractReceipt`.

We must pass the notification handler to the mutation's getter function because otherwise we would have to wait for the transaction to resolve before we would have access to the transaction hash (see the implementation in the getter functions for more details.)

Reply:
i understand why you did it this way, but its really annoying to have to pass this around everywhere you need it. since you are building up an api to call contracts outside of react, why not add a notification "dispatcher" outside of react as well and import it where you need it?  You could do this with a few lines by instantiating an event emitter singleton which could be pulled into any file.  In your notification context subscribe to events, and add notifications from it. In any other place in the app, import the event emitter and emit notification events when you need it.  Then you can effectively eliminate passing around the notification handler.  [Here is an example](https://github.com/UMAprotocol/optimistic-oracle-dapp/blob/master/src/context/OracleClientContext.tsx#L36)

Reply:
this is an awesome solution and I'm excited to implement it. I'll re-request your review once its done.